### PR TITLE
Add dropdown for model selection with the list of compatible models

### LINKED ISSUE #2: Add dropdown for model selection
Please add a dropdown for model selection and default to GPT 3.5 turbo.

Please only add 5 models

### PR DESCRIPTION
**Pull request by aleksandr-efimenko**
Add a dropdown for model selection. Closes #2.
According to the OpenAI [documentation](https://platform.openai.com/docs/models/model-endpoint-compatibility) only gpt-3.5, gpt-4, and their variations are compatible with chat completions. So, I added them. to the list

### CODE REVIEW

File: src/data/available-models.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+// This is a list of all the models that are available to use for the OpenAI API key.
+const availableModels = [
+  "davinci",
```
Comment (cohenoshriconsulting) [inline src/data/available-models.ts:5]: Move this to the database as a migration.
Add the following column s to define their use case

- Available_for_text_chat
- Available_for_image_gen

Comment (cohenoshriconsulting) [reply src/data/available-models.ts:5]: Add additional columns to define the functionality/usecase of the model.
Also add a description column to explain the model as well as a link to the documentation of the model.

File: src/components/openai-playground/chat-playground.tsx [hunk 2]
```diff
@@ -6,15 +6,26 @@ import { Textarea } from "@/components/ui/textarea";
...
   title: "Playground",
   description: "The OpenAI Playground built using the components.",
 };
+const modelList = [
+  "gpt-3.5-turbo",
+  "gpt-3.5-turbo-16k",
+  "gpt-3.5-turbo-0613",
+  "gpt-4",
+  "gpt-4-0613",
+];
```
[the remark this answers is not in the document]
Comment (aleksandr-efimenko) [reply src/components/openai-playground/chat-playground.tsx]: On the front-end side I created query to the api endpoint to get data from the database

`const modelsList = api.openAiModels.getModels.useQuery({
    availableForChat: true,
  }).data;`

File: src/data/available-models.ts (hunk 1 above)
Comment (aleksandr-efimenko) [reply src/data/available-models.ts:5]: I created model for storing openAiModels
`model OpenAIModels {
    name                        String    @id
    availableForChat            Boolean
    availableForImageGeneration Boolean
    availableForEmbeddings      Boolean
    availableForAudioTranscript Boolean
    availableForFineTuningJobs  Boolean
    availableForFineTunes       Boolean
    description                 String?
    maxTokens                   Int?
    TrainingDataDate            DateTime?
    deprecated                  Boolean?
    User                        User?     @relation(fields: [userId], references: [id])
    userId                      String?
}`

Created seed file with 5 models for chat: 
`  const gpt4 = await prisma.openAIModels.upsert({
    where: {
      name: "gpt-4",
    },
    update: {},
    create: {
      name: "gpt-4",
      availableForChat: true,
      availableForImageGeneration: false,
      availableForEmbeddings: false,
      availableForAudioTranscript: false,
      availableForFineTuningJobs: false,
      availableForFineTunes: false,
      description:
        "More capable than any GPT-3.5 model, able to do more complex tasks, and optimized for chat. Will be updated with our latest model iteration 2 weeks after it is released.",
      maxTokens: 8192,
      TrainingDataDate: new Date("2021-09-01"),
    },
  });`

**Review by cohenoshriconsulting — APPROVED** [review]

**Pull request merged**